app/src/main/java/net/squanchy/tweets/service/TweetModelConverter.java

```diff
@@ -120,6 +114,25 @@ private int offsetFrom(int start, List<Integer> indices) {
         return Lists.map(photos, mediaEntity -> mediaEntity.mediaUrlHttps);
     }
 
+    private String displayableTextFor(Tweet tweet, Range displayTextRange, List<MediaEntity> photoUrls) {
+        Integer beginIndex = displayTextRange.start();
+        Integer endIndex = displayTextRange.end();
+        String displayableText = tweet.text.substring(beginIndex, endIndex);
+        if (photoUrls != null) {
```

Comment:
AH the problem is you should be passing in the `List<String> photoUrls` instead, that's guaranteed not to be null. If you need the range metadata create a `PhotoEntity` and change the list of strings to a list of those.